### PR TITLE
Fix #883 Specify client props as HDFS path in new M/R API

### LINKED ISSUE #883: Specify client properties as path in new MapReduce API
The new MapReduce API for 2.0 needs a way to specify client properties as path (that can be cached in HDFS) to provide users with way to avoid serializing properties in Job configuration.

### PR DESCRIPTION
**Pull request by mikewalch**
* Also cleaned up new M/R API by removing unnecessary methods
  and using properties in place of ClientInfo

### CODE REVIEW

File: hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoopImpl/mapreduce/lib/ConfiguratorBase.java [hunk 1]
```diff
@@ -102,69 +78,26 @@ protected static String enumToConfKey(Enum<?> e) {
...
-      Properties props) {
-    StringWriter writer = new StringWriter();
-    try {
-      props.store(writer, "client properties");
-    } catch (IOException e) {
-      throw new IllegalStateException(e);
+      Properties props, String clientPropsPath) {
+    if (clientPropsPath != null) {
+      try {
+        DistributedCacheHelper.addCacheFile(new URI(clientPropsPath), conf);
```
Comment (milleruntime) [inline hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoopImpl/mapreduce/lib/ConfiguratorBase.java]: Should update the DistributedCacheHelper to not use deprecated methods so it can be un-deprecated.  This could be a separate PR though

File: hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoop/mapreduce/InputFormatBuilder.java [hunk 2]
```diff
@@ -45,14 +45,26 @@
...
      *          Accumulo connection information
      */
     TableParams<T> clientProperties(Properties clientProperties);
+
+    /**
+     * Set path to HDFS location containing accumulo-client.properties file. This setting is more
+     * secure than {@link #clientProperties(Properties)}
+     *
+     * @param clientPropsPath
+     *          HDFS path to accumulo-client.properties
```
Comment (milleruntime) [inline hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoop/mapreduce/InputFormatBuilder.java]: This could be from a DFS location and doesn't have to be HDFS right?

File: hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoopImpl/mapreduce/lib/ConfiguratorBase.java (hunk 1 above)
Comment (mikewalch) [reply hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoopImpl/mapreduce/lib/ConfiguratorBase.java]: I need something that works with both mapreduce & mapred.  I am not sure what a better option is.  We might be able to do the following but I think this should be a separate issue (I will create one):
```java
    String files = conf.get("mapreduce.job.cache.files");
    conf.set("mapreduce.job.cache.files", files == null ? uri.toString() : files + "," + uri.toString());
```

File: hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoop/mapreduce/InputFormatBuilder.java (hunk 2 above)
Comment (mikewalch) [reply hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoop/mapreduce/InputFormatBuilder.java]: Yes it doesn't have be HDFS.  I update the docs.
